app/controllers/requests_controller.rb

```diff
@@ -23,6 +24,12 @@ def current_request
 
   protected
 
+  def check_if_proxy_sponsor
+    return if params[:proxy].present? || !current_user.sponsor?
```

Comment:
I'm receiving an error that traces back to this line when submitting a page request with a library ID.  (I guess that means we've been missing an important integration test about submitting w/ Library ID only).

In the case when a request is submitted w/ a Library ID only (or name+email) the `current_user` is still anonymous.  The user object that *I think* we're interested in is accessible through the `current_request`.